eventuate-local-java-embedded-cdc/src/test/java/io/eventuate/local/cdc/debezium/AbstractTopicRelayTest.java

```diff
@@ -66,8 +73,41 @@ public void shouldCaptureAndPublishChange() throws ExecutionException, Interrupt
     long endTime = System.currentTimeMillis();
 
     logger.debug("got the event I just published in msecs {}", endTime - publishTime);
+  }
+
+  @Test
+  public void testDelete() throws ExecutionException, InterruptedException {
 
-//    eventTableChangesToAggregateTopicRelay.stopCapturingChanges();
+    String aggregateType = "TestAggregate" + UUID.randomUUID().toString();
+    String eventType = "TestEvent" + UUID.randomUUID().toString();
+
+    List<EventTypeAndData> myEvents = Collections.singletonList(new EventTypeAndData(eventType, "{}", Optional.empty()));
+
+    EntityIdVersionAndEventIds ewidv = AsyncUtil.await(eventuateJdbcEventStore.save(aggregateType, myEvents, Optional.empty()));
+
+    Set<Int128> expectedEventIds = new HashSet<>();
+    expectedEventIds.add(ewidv.getEntityVersion());
+    BlockingQueue<Int128> result = new LinkedBlockingDeque<>();
+    eventuateKafkaAggregateSubscriptions.subscribe("testSubscriber-" + getClass().getName(),
+            Collections.singletonMap(aggregateType, Collections.singleton(eventType)),
+            SubscriberOptions.DEFAULTS,
+            se -> {
+              logger.debug("got se {}", se);
+              if (expectedEventIds.contains(se.getId()))
+                result.add(se.getId());
+              return CompletableFuture.completedFuture(null);
+            }).get();
+
+    Assert.assertNotNull(result.poll(30, TimeUnit.SECONDS));
+    Assert.assertNull(result.poll(30, TimeUnit.SECONDS));
+
+    new JdbcTemplate(dataSource).update(String.format("delete from %s", eventuateSchema.qualifyTable("events")));
+
+    ewidv = AsyncUtil.await(eventuateJdbcEventStore.save(aggregateType, myEvents, Optional.empty()));
+    expectedEventIds.add(ewidv.getEntityVersion());
```

Comment:
There is a (slight) possibility that the subscriber has already received and discarded the event before telling it to look.

I think it would be cleaner to subscribe() for the second time.

Reply:
ok